### PR TITLE
Feature/submitter confirms upload gdev 283

### PR DESCRIPTION
**Pull request by MoritzHahn1337**
Title for squash and merge:
```
Data Submitter confirms the upload of files (GDEV-283 & GDEV-285)
```

Description for squash and merge:
```
Added /confirm_upload/{file_id} endpoint for the uploader to confirm the upload
Return 404 if the file is not in the inbox and 400 if the file_id is not in the database, 200 else
Publishes file_upload_received if the upload was successfull
```

### CODE REVIEW

**Comment by MoritzHahn1337** [comment]
@KerstenBreuer do you know how we could fix the openapi-checker error?

File: openapi.yaml [hunk 1]
```diff
@@ -34,6 +34,33 @@ info:
   version: 0.1.0
 openapi: 3.0.2
 paths:
+  /confirm_upload/{file_id}:
```
Comment (KerstenBreuer) [inline openapi.yaml:37]: Die API, können wir später noch etwas überarbeiten, um sie mehr "RESTful" zu machen. Aber das muss den aktuellen PR nicht aufhalten.

File: upload_controller_service/api/main.py [hunk 2]
```diff
@@ -56,3 +57,30 @@ async def get_presigned_post(
...
+            status_code=400,
+            detail=f"The submitted file_id {file_id} does not exist.",
+        ) from file_info_not_found_error
+    except ObjectNotFoundError as object_not_found_error:
+        raise HTTPException(
+            status_code=404,
+            detail=f"The file with the file_id {file_id} does not exist.",
+        ) from object_not_found_error
+
+    return True
```
Comment (KerstenBreuer) [inline upload_controller_service/api/main.py]: Returning a 204 might be a bit better here. Something like:
```suggestion
@app.get("/confirm_upload/{file_id}", summary="confirm_upload", status_code=HTTP_204_NO_CONTENT)
async def confirm_upload(
    file_id: str,
    config: Config = Depends(get_config),
):
    """
    Requesting a confirmation of the upload of a specific file using the file id.
    Returns 204, if the file exists in the inbox, 404 if not
    """

    # call core functionality
    try:
        check_uploaded_file(file_id=file_id, config=config)
    except FileInfoNotFoundError as file_info_not_found_error:
        raise HTTPException(
            status_code=400,
            detail=f"The submitted file_id {file_id} does not exist.",
        ) from file_info_not_found_error
    except ObjectNotFoundError as object_not_found_error:
        raise HTTPException(
            status_code=404,
            detail=f"The file with the file_id {file_id} does not exist.",
        ) from object_not_found_error

    return Response(status_code=HTTP_204_NO_CONTENT)
```

File: upload_controller_service/api/main.py [hunk 3]
```diff
@@ -56,3 +57,30 @@ async def get_presigned_post(
...
+    config: Config = Depends(get_config),
+):
+    """
+    Requesting a confirmation of the upload of a specific file using the file id.
+    Returns 200, if the file exists in the inbox, 404 if not
+    """
+
+    # call core functionality
+    try:
+        check_uploaded_file(file_id=file_id, config=config)
```
Comment (KerstenBreuer) [inline upload_controller_service/api/main.py]: This function is nice, however, shouldn't we also track that the file has now been declared as "uploaded"?

File: upload_controller_service/core/main.py [hunk 4]
```diff
@@ -53,3 +59,24 @@ def get_upload_url(file_id: str, config: Config = CONFIG):
...
+        if not storage.does_bucket_exist(bucket_id=config.inbox_bucket_name):
+            raise BucketNotFoundError
+
+        if not storage.does_object_exist(
+            object_id=file_id,
+            bucket_id=config.inbox_bucket_name,
+        ):
+            raise ObjectNotFoundError
+
+    publish_upload_received(file=file, config=config)
```
Comment (KerstenBreuer) [inline upload_controller_service/core/main.py]: This function should be given as an argument to the check_upload_file function. Just like here: https://github.com/ghga-de/drs3/blob/dev/drs3/core/main.py#L34

**Review by KerstenBreuer — CHANGES_REQUESTED** [review]

File: upload_controller_service/api/main.py (hunk 3 above)
Comment (KerstenBreuer) [reply upload_controller_service/api/main.py]: Will be done in another PR.

Comment (MoritzHahn1337) [reply upload_controller_service/api/main.py]: https://jira.verbis.dkfz.de/browse/GDEV-478

File: openapi.yaml (hunk 1 above)
Comment (MoritzHahn1337) [reply openapi.yaml:37]: https://jira.verbis.dkfz.de/browse/GDEV-479

**Review by KerstenBreuer — APPROVED** [review]

**Pull request merged**